cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -28,6 +28,44 @@
 
 KUBECTL=${KUBECTL_BIN:-/usr/local/bin/kubectl}
 KUBECTL_OPTS=${KUBECTL_OPTS:-}
+# KUBECTL_PRUNE_WHITELIST is a list of resources whitelisted by
+# default.
+# This is currently the same with the default in:
+# https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/apply.go
+KUBECTL_PRUNE_WHITELIST=(
```

Comment:
The problem is that when you need to specify one extra resource, you need to specify the whole list.

Think about if you are a vendor, and you need a new resource, you have to specify the whole list -- a copy of OSS default resource list + the extra resource. This means that all vendors who need extra prune resource will have to maintain a copy of the default list, which is >> 2 places.

And actually I think it makes sense to maintain a separate list for the addon manager. The default list in `kubectl apply --prune` is just a general default for all use cases, it is not designed for the addon manager. When we need to remove or add a default prune resource for the addon manager, I don't think we can change the `kubectl` default, because that will be a breaking change for the `kubectl` users.
 
However, if we have a separate list for the addon manager, it can be very flexible, because addon manager is the only user. Even today, we can extend/reduce this list to make it a better fit for addon manager.

Reply:
Thanks for the explanation, that makes more sense now. I scanned too quick and didn't realize you are adding "extra whitelists".

Another thought would be letting `kubectl apply` to have the ability to take extra whitelists, though given the state of this feature, it might not worth it. Would really hope we can get rid of these whole thing soonish :)